### PR TITLE
Fix(script/LBRS)  urok doomhowl improvements

### PR DESCRIPTION
**Pull request by patou01**
"Urok enforcer" and "Urok ogre magus" (entry 10601, 10602) appear neutral.
They should be hostile. Fixed.
Urok has wrong yell on aggro "me smash, you die". Should be fixed to his yell.
Urok was not casting intimidating shout. Added. Timer is TBD

Partial fix for AzerothCore 7364

## Changes Proposed:
- Add intimidating shout to the casts on timer.
- changed Urok enforcer and Urok ogre mage faction (making them hostile at the same time). Removed civilian flag as well.
- move the "say" to initialize. Added an on-spawn aggro of nearest target.

## Issues Addressed:

- Partial fix of https://github.com/azerothcore/azerothcore-wotlk/issues/7364
- partial fix for https://github.com/azerothcore/azerothcore-wotlk/issues/8712 , not sure the despawn/resummon part is a bug. I can resummon him, but you should only ever get one head in the dungeon...

## SOURCE:

https://youtu.be/AY-m_3lGiqg?t=97 . It's in the issue too.

## Tests Performed:

- Builds. 
- Fear is casted. 
- Urok yells on spawn.
- Other mobs appear red
- Urok attacks on spawn

## How to Test the Changes:

1. Build and apply changes
2. go to lbrs
.additem 12533
.additem 12534
.go o 99791
Fight through event
3. verify changes

1. start the event but stay in gm mode (or stealthed, or anything so the mobs don't aggro).
2. The boss should eventually despawn, and do so before the mobs despawn.

## Known Issues and TODO List:

- [ ] mobs attacking the pike
- [ ] Is the timer ok? Do we have any source on the time it should be?

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_urok_doomhowl.cpp [hunk 1]
```diff
@@ -47,14 +47,19 @@ class boss_urok_doomhowl : public CreatureScript
...
+                me->AI()->AttackStart(target);
+            }
         }
 
         void EnterCombat(Unit* /*who*/) override
         {
             _EnterCombat();
             events.ScheduleEvent(SPELL_REND, urand(17000, 20000));
             events.ScheduleEvent(SPELL_STRIKE, urand(10000, 12000));
-            Talk(SAY_AGGRO);
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_urok_doomhowl.cpp:57]: why was this removed?

Comment (Kitzunu) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_urok_doomhowl.cpp:57]: if it is the wrong yell, you need to fix it in creature_text instead of just deleting it

Comment (patou01) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_urok_doomhowl.cpp:57]: My understanding is that he has no on_combat yell. No boss will start a fight with "me smash you die" like he's a random ogre. And his yell is "You have **summoned** me, fool! Now Die!". See here https://youtu.be/jeTFFQ37tW0?t=8  
So I put that yell on the on_summon, and since I understood it that he should engage on summon, then there should be no on_combat?  It's true that he might say something when he is engaged after a reset, but I don't know if we have any source on that? Or does he despawn on wipe?

Here they suggest he does a yell and a say in sequence, basically erasing the yell. https://youtu.be/W7YdNcfEDgo?t=204 That makes no sense and can only be an error.

Comment (Kitzunu) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_urok_doomhowl.cpp:57]: I'd put the other text in EnterCombat instead of OnSummon just incase of resets etc

Comment (patou01) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_urok_doomhowl.cpp:57]: The yell specifically mentions summoning. Ogres are stupid, but I feel that would be worse than having nothing. We can ask around and see if someone could run it on classic and vanish or something to see what happens

1. When he resets (despawn?)
2. on engage without summon (does he yell?)

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by patou01** [comment]
In my opinion ready for review/test, though @kitzunu disagreed with removing the yell. I don't have a way of testing on retail and having a boss having the same say on aggro as mobs makes no sense to me.

**Comment by temperrr** [comment]
If there is more testing needed, feel free to add the label back :)

**Review by UltraNix — DISMISSED** [review]

**Comment by temperrr** [comment]
Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK1181	cannot open input file '..\scripts\RelWithDebInfo\scripts.lib'	worldserver	F:\Build\src\server\worldserver\LINK	1	
Error	C2660	'CreatureAI::DoZoneInCombat': function does not take 2 arguments	scripts	H:\Azerothcore\src\server\scripts\EasternKingdoms\BlackrockMountain\BlackrockSpire\boss_urok_doomhowl.cpp	62	


When building with linked PR

**Comment by patou01** [comment]
My bad. Since the discussion on the linked PR stopped at "I'll think about it" on a suggestion, I thought we would put this one in and fix DoZoneInCombat part later as it means changing a single line and the impact on behaviour would be minimal.

**Comment by temperrr** [comment]
Ok Ill test without that PR then

**Comment by temperrr** [comment]
Timer is OK
I feel like the despawn time is too long, shouldn't the event just reset the moment they can't find a target in combat ?
All summoned monsters drop food & drink (probably outside scope of this pr)

**Comment by patou01** [comment]
> Timer is OK I feel like the despawn time is too long, shouldn't the event just reset the moment they can't find a target in combat ? 

I can see if I find something in SAI to do that, since the mobs and events are in SAI it's a bit hard to synchronize eveything... Else I can arguably move it to c++, but then I need to write the spell part of the pike. I'll look into it.

> All summoned monsters drop food & drink (probably outside scope of this pr)

Should they drop anything at all? If not, I can drop that easily.

> Also noting that runes should disappear when monster is spawned.

after X seconds or something? Since the runes are re-used, or are we spawning too many mobs?

> Maybe we should merge and fix the remaining issues in a new PR?

Depends on the complexity of the tasks. I'll have a look into the code during the week and see what I can propose to improve that.

**Comment by patou01** [comment]
The latest commit should fix most issues, I just need to check if the boss doesn't despawn after 30s even in combat, I have a doubt on the summon flag

**Comment by patou01** [comment]
@temperrr, I reworked the reset a bit, though I don't know if I'm supposed to respawn the tribute (or whatever the gameobject is that is used to spawn the boss). 
Could you clarify the loot issue? Maybe I can squeeze that one in...

**Comment by Nyeriah** [comment]
I believe we should merge this and fix any remaining issues in a new PR. What do you think @patou01 ?

PRs don't need to fix every issue there is, they just need to address the issues they propose to solve and from what I see those were already taken care of.

Thanks for all the hard work

**Comment by Nefertumm** [comment]
Fix the merge conflicts please :D

**Comment by patou01** [comment]
😭. Will do tonight. Someone fixed something else in BRS. Definitely need more modularity to avoid constant merge conflicts.

**Comment by Nyeriah** [comment]
Merge conflict resolved, moving to-be-merged

**Comment by patou01** [comment]
I assume it's re-tested before going to chromie? As I reworked the reset logic quite a lot it could be good to have a look.

**Comment by Nyeriah** [comment]
> I assume it's re-tested before going to chromie? As I reworked the reset logic quite a lot it could be good to have a look.

Yes, we are retesting the entire instance tonight
